### DISCUSSION TITLE
How to fetch common dynamic data efficiently?

### DISCUSSION THREAD

**Discussion by deebov** [vercel/next.js#26807] [Help]
Hey all 👋

I have a case where on every page I fetch a data about sidebar and render it. The sidebar looks same on all pages.

This is where fetch the data: 
`pages/[...slug].tsx`

I dont need the slug in order to fetch the data but I need the slug to extract the currect page from the fetched sidebar data. So right now I believe nextjs is fetch the sidebar when generating static pages from `getStaticPaths`.

Is there a way to fetch the sidebar data only once when building the static pages?

**Comment by bitroy** [comment]
Check this example -> [layout-component](https://github.com/vercel/next.js/tree/canary/examples/layout-component)
